### PR TITLE
Updates from 3.5.1 release

### PR DESCRIPTION
**Pull request by metanivek**
This PR is bringing changes that came up during the 3.5.1 release (https://github.com/mirage/irmin/pull/2162), either in testing or during the opam CI process.

### CODE REVIEW

File: src/irmin-pack/unix/checks.ml [hunk 1]
```diff
@@ -200,9 +200,10 @@ module Make (Store : Store) = struct
                 | Error (`Msg m) -> Fmt.kstr Lwt.fail_with "Invalid hash %S" m)
               heads
       in
-      let+ result =
+      let* result =
         Store.integrity_check ~ppf:Format.err_formatter ~auto_repair ~heads repo
       in
+      let+ () = Store.Repo.close repo in
```
Comment (art-w) [inline src/irmin-pack/unix/checks.ml:206]: Nice catch!

**Comment by adatario** [comment]
For future reference this is why tests are disabled on macos: https://github.com/ocaml/opam-repository/pull/22837#issuecomment-1376163293

File: src/irmin-pack/unix/checks.ml (hunk 1 above)
Comment (metanivek) [reply src/irmin-pack/unix/checks.ml:206]: I was only the bug fixer here -- Victor found it during his testing!

**Comment by metanivek** [comment]
@adatario thanks for adding that link. For more context, the tests were failing for the same reason that `libirmin` fails, namely [this bug](https://github.com/mirage/ca-certs/issues/20). It has been a constant source of headache for us. If I had a mac I would definitely look into fixing it.

**Review by clecat — COMMENTED** [review]
It looks good to me, thx for your work

**Pull request merged**